### PR TITLE
correct linting errors for typelist/max_items schema attributes

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #9956 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
correct typeset usage with typelist in cases where max_items is 1
```

Output from acceptance testing:

```
PASS: TestAccAWSFsxWindowsFileSystem
PASS: TestAccAWSFsxLustreFileSystem
PASS: TestAccAWSAPIGatewayUsagePlan
PASS: TestAccAWSCloudFrontDistribution
PASS: TestAccAWSAPIGatewayVpcLink
PASS: TestAccAWSEcsService
PASS: TestAccAWSEcsServiceDataSource
PASS: TestAccAWSS3Bucket
PASS: TestAccAWSS3BucketMetric
PASS: TestAWSS3BucketName 
PASS: TestAccAWSS3BucketInventory
PASS: TestAccAWSS3BucketPolicy
PASS: TestAccAWSS3BucketPublicAccessBlock
PASS: TestAccAWSCodeBuildProject
PASS: TestAccAWSWafByteMatchSet
PASS: TestAccAWSS3BucketObject
PASS: TestAWSWafSqlInjectionMatchSet
PASS: TestAWSWafWebAcl
PASS: TestAccAWSWafByteMatchSet
PASS: TestAccAWSS3BucketNotification
PASS: TestAccAWSVPCPeeringConnection
PASS: TestAccAWSVPCPeeringConnectionAccepter
PASS: TestAccAWSS3BucketPublicAccessBlock
PASS: TestAccAWSS3BucketAnalyticsConfiguration
PASS: TestAccAWSElasticTranscoderPipeline
PASS: TestAccAWSElasticTranscoderPreset
PASS: TestWebsiteEndpoint
PASS: TestBucketRegionalDomainName
PASS: TestAccAWSKinesisFirehoseDeliveryStream
PASS: TestAccAWSSESEventDestination
```
```
FAIL: TestAccAWSS3Bucket_LifecycleRule_Expiration_EmptyConfigurationBlock
 ```

### CODE REVIEW

File: aws/resource_aws_s3_bucket.go [hunk 1]
```diff
@@ -2191,33 +2186,38 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
 		}
 
 		// Expiration
-		expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.expiration", i)).(*schema.Set).List()
+		expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.expiration", i)).([]interface{})
```
Comment (anGie44) [inline aws/resource_aws_s3_bucket.go:2189]: wondering if this is the correct way to get the `expiration` value from the resource b/c I noticed if expiration is passed in as empty `expiration {}`  block for example in a TF config, the value of this `expiration` variable is `[ nil ]`, requiring the 2 `if` checks below..while previously only 1 `if` was needed

File: aws/resource_aws_s3_bucket.go [hunk 2]
```diff
@@ -2191,33 +2186,38 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
...
-			} else if val, ok := e["expired_object_delete_marker"].(bool); ok {
-				i.ExpiredObjectDeleteMarker = aws.Bool(val)
+				rule.Expiration = i
 			}
-			rule.Expiration = i
 		}
 
 		// NoncurrentVersionExpiration
-		nc_expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.noncurrent_version_expiration", i)).(*schema.Set).List()
+		nc_expiration := d.Get(fmt.Sprintf("lifecycle_rule.%d.noncurrent_version_expiration", i)).([]interface{})
```
Comment (anGie44) [inline aws/resource_aws_s3_bucket.go:2212]: same comment as above

File: aws/resource_aws_waf_sql_injection_match_set.go [hunk 3]
```diff
@@ -104,6 +104,7 @@ func resourceAwsWafSqlInjectionMatchSetRead(d *schema.ResourceData, meta interfa
 	if err := d.Set("sql_injection_match_tuples", flattenWafSqlInjectionMatchTuples(resp.SqlInjectionMatchSet.SqlInjectionMatchTuples)); err != nil {
 		return fmt.Errorf("error setting sql_injection_match_tuples: %s", err)
 	}
+	log.Printf("Set to: %v", d.Get("sql_injection_match_tuples").(*schema.Set))
```
Comment (anGie44) [inline aws/resource_aws_waf_sql_injection_match_set.go:107]: ah a debug log..will remove

File: aws/resource_aws_s3_bucket.go (hunk 1 above)
Comment (anGie44) [reply aws/resource_aws_s3_bucket.go:2189]: value extracted here and in `nc_expiration` at L2212 could also explain new test failure seen in ` TestAccAWSS3Bucket_LifecycleRule_Expiration_EmptyConfigurationBlock`...though following the logic, the error seems to makes sense to me as nothing in the `rule` object will be populated in terms of an `action` if one were to pass just the `enabled` and `id` params to `lifecycle_rule`

**Comment by bflad** [comment]
Not sure how @gdavison feels on the subject, but can you please split this up per resource or service? Some of these are going to be a lot harder to review/fix than others and also for some of these we can close out a laundry list of issues (e.g. `aws_codebuild_project`), so it would be nice to iteratively get these fixes in. Thanks so much!

**Comment by anGie44** [comment]
sure thing! I was just starting to pull out 3 resources b/c of test failures and was thinking to myself I should of split things up more 😅

**Pull request closed without merging**